File: newFiveAngular/src/app/model/reservation.ts
import { Article } from './article';
import { TerrainReserve } from './terrainReserve';
import { Time } from "@angular/common";

export class Reservation extends Article{
  constructor(


    private _terrainReserve=new TerrainReserve,
    private _date?: Date,
    private _heure?: Time,

  ) {
    super()

  }


  public get terrainReserve(): TerrainReserve {
    return this._terrainReserve;
  }
  public set terrainReserve(value: TerrainReserve) {
    this._terrainReserve = value;
  }

  public set date(value: Date | undefined) {
    this._date= value;
  }
  public get date() {
    return this._date;
  }
  public set heure(value: Time | undefined) {
    this._heure= value;
  }
  public get heure(): Time| undefined {
    return this._heure;
  }

}

File: newFive/src/main/java/formation/projetParis/newFive/entities/Reservation.java
package formation.projetParis.newFive.entities;

import java.time.LocalDate;
import java.time.LocalTime;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("R")

public class Reservation extends Article {


	@JsonView(JsonViews.Base.class)
	private LocalDate date;
	@JsonView(JsonViews.Base.class)
	private LocalTime heure;
	@OneToOne
	@JoinColumn(name = "terrain_number") //nom dans la base
	@JsonView(JsonViews.Base.class)
	private Terrain terrainReserve;
	
	
	public Reservation(LocalDate date, LocalTime heure) {

		this.date = date;
		this.heure = heure;
	}

	public Reservation() {
		super();
	}

	public LocalDate getDate() {
		return date;
	}

	public void setDate(LocalDate date) {
		this.date = date;
	}

	public LocalTime getHeure() {
		return heure;
	}

	public void setHeure(LocalTime heure) {
		this.heure = heure;
	}

	public Terrain getTerrainReserve() {
		return terrainReserve;
	}

	public void setTerrainReserve(Terrain terrainReserve) {
		this.terrainReserve = terrainReserve;
	}
	
	


}